subsys/net/lib/mcumgr_smp_client/src/mcumgr_smp_client.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <ctype.h>
+#include <zephyr/drivers/gpio.h>
+#include <stdio.h>
+#include <dfu/dfu_target_smp.h>
+#include <net/fota_download.h>
+#include <fota_download_util.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(MCUmgr_smp_client, CONFIG_NRF_MCUMGR_SMP_CLIENT_LOG_LEVEL);
+
+static dfu_target_reset_cb_t reset_cb;
+
+int mcumgr_smp_client_init(dfu_target_reset_cb_t cb)
+{
+	int ret;
+
+	ret = dfu_target_smp_client_init();
+	if (ret) {
+		LOG_ERR("Failed to init DFU target SMP, %d", ret);
+		return ret;
+	}
+
+	dfu_target_smp_recovery_mode_enable(cb);
+	reset_cb = cb;
+
+	return ret;
+}
+
+int mcumgr_smp_client_download_start(const char *download_uri, int sec_tag,
+				     fota_download_callback_t client_callback)
+{
+	return fota_download_util_download_start(download_uri, DFU_TARGET_IMAGE_TYPE_SMP, sec_tag,
+						   client_callback);
+}
+
+int mcumgr_smp_client_download_cancel(void)
+{
+	return fota_download_util_download_cancel();
+}
+
+int mcumgr_smp_client_update(void)
+{
+	int ret;
+
+	ret = fota_download_util_image_schedule(DFU_TARGET_IMAGE_TYPE_SMP);
+
+	return ret;
```

Comment:
Does this really need a ret value if it just calls a function without doing anything else?